src/post-submission-actions/program-enrollment-action.ts

```diff
@@ -33,8 +33,14 @@ export const ProgramEnrollmentSubmissionAction: PostSubmissionAction = {
             (enrollment) => enrollment.program.uuid === programUuid && enrollment.dateCompleted === null,
           );
           if (hasActiveEnrollment) {
-            throw new Error('Cannot enroll patient to program. Patient already has an active enrollment');
+            showToast({
+              title: 'Error saving enrollment',
```

Comment:
Can we use translatable strings?

Reply:
Form the current implementation [here](https://github.com/openmrs/openmrs-form-engine-lib/blob/daea8e29d28dd2ec0cd1af684123329320332bb8/src/components/encounter/encounter-form.component.tsx#L594) it seems that saving an encounter is dependent on saving an identifier

Reply:
Is that the regression? @CynthiaKamau I understand the encounter isn't related to the patient identifiers from a modelling pov but I would keep it that way. What happens if the patient ID fails to save due to a validation error? In such scenarios we want to halt the submission and present the error(s) the user. It's possible for the user to not only fix the issues related to the ID but also make changes to the encounter specific fields; imagine if you had already submitted the encounter! you would have to edit the encounter (this requires some work with the current state of things) or may end up posting another encounter (most likely to happen). To prevent this mess, it makes sense to keep it that.

Reply:
Makes sense

Reply:
Done